server/cli_common/utils.py

```diff
@@ -30,7 +39,162 @@ def get_stain_matrix(args, count=3):
     Return a numpy array of column vectors.
 
     """
-    return numpy.array([get_stain_vector(args, i+1) for i in range(count)]).T
+    return np.array([get_stain_vector(args, i+1) for i in range(count)]).T
+
+
+def segment_wsi_foreground_at_low_res(ts):
+
+    ts_metadata = ts.getMetadata()
+
+    # get image at low-res
+    maxSize = max(ts_metadata['sizeX'], ts_metadata['sizeY'])
+
+    downsample_factor = 2 ** np.floor(np.log2(maxSize / 2048))
+
+    fgnd_seg_mag = ts_metadata['magnification'] / downsample_factor
+
+    fgnd_seg_scale = {'magnification': fgnd_seg_mag}
+
+    im_lres, _ = ts.getRegion(
+        scale=fgnd_seg_scale,
+        format=large_image.tilesource.TILE_FORMAT_NUMPY
+    )
+
+    im_lres = im_lres[:, :, :3]
+
+    # compute foreground mask at low-res
+    im_fgnd_mask_lres = htk_utils.simple_mask(im_lres)
+
+    return im_fgnd_mask_lres, fgnd_seg_scale
+
+
+def detect_nuclei_kofahi(im_nuclei_stain, args):
```

Comment:
Would `detect_nuclei_kofahi` be better in the core package? Except for taking `args` instead of individual parameters, it doesn't seem to do anything CLI-specific.